packages/next/next-server/lib/fid.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * This is a modified version of the First Input Delay polyfill
+ * https://github.com/GoogleChromeLabs/first-input-delay
```

Comment:
could we reuse this code and only override what's necessary i.e. override reportFirstInputDelayIfRecordedAndValid()?

Reply:
Not sure if that's possible 🤔

Also a few things were modified throughout the file (naming, etc...) so maybe it's better to have all the logic encapsulated here?